fix(Friends): Clear friend adding error whenever we leave the friends…

InfamousVague opened the pull request:
… tab

<!--  Thanks for sending a pull request!-->

### What this PR does 📖
- 

### Which issue(s) this PR fixes 🔨
- Resolve #
<!--Add the ticket Github number such as #Resolve #001 to automatically link the PR to the issue-->

### Special notes for reviewers 🗒️


### Additional comments 🎤

InfamousVague committed:
fix(Friends): Clear friend adding error whenever we leave the friends tab

stavares843 reviewed (commented):
i deleted the cache folder > checkout this branch > cargo run > added a friend > tried to add again > error > left that section > went there again > still shows the error


https://user-images.githubusercontent.com/29093946/193431102-060ebfc4-4cc7-462f-98b9-fd84d1c0d40b.mov

InfamousVague commented:
> i deleted the cache folder > checkout this branch > cargo run > added a friend > tried to add again > error > left that section > went there again > still shows the error
> Gravacao.do.ecra.2022-10-01.as.23.49.56.mov

You can ignore that for now since it's not critical the error is returned when you interact with the friend in some way or change the text inside the input field. It is strange that it's showing up above the input though because I thought I moved that down below. 

Regaurdless we should make a new backlogged ticket to re-render that component and have it actually unmount once the animation to slide off screen is finished. Right now it's got an "always on" kinda setup in the UI but realistically it should have a delay then make closure call to request it be un rendered (i.e. putting a stateful conditional on the parent component and only render friends (and other modals) when they haven't requested they be destroyed, then we can have the parent component manage the opening and closing of the widget (should be a minor change to existing code)).
